pkg/proxy/transport/transport.go

```diff
@@ -53,8 +68,15 @@ func New(opts ...Option) *http.Transport {
 		t.idleConnectionsPerHost = DefaultIdleConnsPerHost
 	}
 
-	if t.closeIdleConnsPeriod == 0 {
-		t.closeIdleConnsPeriod = DefaultCloseIdleConnsPeriod
+	if t.idleConnTimeout == 0 {
+		t.idleConnTimeout = DefaultIdleConnTimeout
+	}
+
+	// let's try to get the cached transport from registry, since there is no need to create lots of
+	// transports with the same configuration
+	hash := t.hash()
+	if tr, ok := registry[hash]; ok {
```

Comment:
I think this might cause race conditions, right? If we create transactions inside goroutines, this can break. Did you run it with the `--race` flag to check?

Reply:
Good catch! No, I did not check it, will do and add thread-safe for registry.